### ISSUE TITLE
pdz.py doesn't work on the main.pdz from settings

### ISSUE THREAD

**Issue by EngineerRunner** [cranksters/playdate-reverse-engineering#23]
i tried using pdz.py on the main.pdz file found in settings, but i got this error.

Traceback (most recent call last):
  File "C:\Users\user\Documents\reverse-playdate\pdz.py", line 114, in <module>
    pdz.save_entries(argv[2])
  File "C:\Users\user\Documents\reverse-playdate\pdz.py", line 101, in save_entries
    self.save_entry_data(name, outdir)
  File "C:\Users\user\Documents\reverse-playdate\pdz.py", line 92, in save_entry_data
    data = self.get_entry_data(name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Documents\reverse-playdate\pdz.py", line 86, in get_entry_data
    return decompress(entry['data'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^
zlib.error: Error -3 while decompressing data: incorrect header check

i did get this file from simulator and not real hardware, so that might be it.

**Comment by scratchminer** [comment]
I think this is because `pdz.py` was written before we knew about the fact that PDA files in a PDZ have extra header fields.

**Comment by melomac** [comment]
I have been trying to reverse the `main.pdz` from the Playdate itself for educational purpose only:

```text
/Games/Seasons/Season-001/Whitewater Wipeout.pdx/main.pdz
```

and I ended up with the same error:

```text
Traceback (most recent call last):
  File "/usr/local/src/playdate-reverse-engineering/tools/pdz.py", line 114, in <module>
    pdz.save_entries(argv[2])
  File "/usr/local/src/playdate-reverse-engineering/tools/pdz.py", line 101, in save_entries
    self.save_entry_data(name, outdir)
  File "/usr/local/src/playdate-reverse-engineering/tools/pdz.py", line 92, in save_entry_data
    data = self.get_entry_data(name)
  File "/usr/local/src/playdate-reverse-engineering/tools/pdz.py", line 86, in get_entry_data
    return decompress(entry['data'])
zlib.error: Error -3 while decompressing data: incorrect header check
```

Moving forward, will this project gets updated to support current Playdate OS and games?

**Comment by gatesphere** [comment]
"fixed" this in pr #27 just submitted.

**Comment by castholm** [comment]
Fixed by #28.
